scripts/o3de/o3de/repo_properties.py

```diff
@@ -306,12 +306,15 @@ def edit_repo_props(repo_path: pathlib.Path = None,
         repo_json['repo_name'] = repo_name
 
     if add_gems or delete_gems or replace_gems:
+        utils.backup_file(repo_path)
         _edit_objects('gem', validation.valid_o3de_gem_json, repo_json, add_gems, delete_gems, replace_gems, release_archive_path, force, download_prefix)
 
     if add_projects or delete_projects or replace_projects:
+        utils.backup_file(repo_path)
```

Comment:
Rather than placing three backup calls inside each if block , I recommend you save a copy of `repo_json` at the start of the function and then at the end of this function on line 319 you call `utils.backup_file` only if the contents changed like this:
```
if repo_json_original != repo_json:
    utils.backup_file(repo_path)
```

There a case when the backup file could overwrite an existing backup when you don't want it to - for example, say you already have a backup file and you run this function with a syntax error that causes _edit_objects() to fail, your backup file would get replaced even though no changes were made to the file.

Also, the backup file would be created multiple times if you combined add and remove commands.